pkg/solana/client/multinode_client.go

```diff
@@ -45,228 +43,60 @@ var _ mn.SendTxRPCClient[*solana.Transaction, *SendTxResult] = (*MultiNodeClient
 
 type MultiNodeClient struct {
 	Client
-	cfg         *config.TOMLConfig
-	stateMu     sync.RWMutex // protects state* fields
-	subsSliceMu sync.RWMutex
-	subs        map[mn.Subscription]struct{}
-
-	// chStopInFlight can be closed to immediately cancel all in-flight requests on
-	// this RpcClient. Closing and replacing should be serialized through
-	// stateMu since it can happen on state transitions as well as RpcClient Close.
-	chStopInFlight chan struct{}
-
-	chainInfoLock sync.RWMutex
-	// intercepted values seen by callers of the rpcClient excluding health check calls. Need to ensure MultiNode provides repeatable read guarantee
-	highestUserObservations mn.ChainInfo
-	// most recent chain info observed during current lifecycle (reseted on DisconnectAll)
-	latestChainInfo mn.ChainInfo
+	*mn.MultiNodeClient[rpc.Client, *Head]
+	cfg *config.TOMLConfig
 }
 
 func NewMultiNodeClient(endpoint string, cfg *config.TOMLConfig, requestTimeout time.Duration, log logger.Logger) (*MultiNodeClient, error) {
```

Comment:
It's confusing to have two multiNode clients. Is there a better name for one or the other?

Reply:
Good point, I'll rename the MultiNode side to `MultiNodeAdaptor` since it's used to integrate into existing chain-specific clients.